src/java.base/share/classes/java/io/Closeable.java

```diff
@@ -50,4 +52,35 @@ public interface Closeable extends AutoCloseable {
      * @throws IOException if an I/O error occurs
      */
     public void close() throws IOException;
+
+    /**
+     * Closes this stream and releases any system resources associated
+     * with it. If the stream is already closed then invoking this
+     * method has no effect.
+     *
```

Comment:
Can you add an example showing the recommended pattern of use?

Reply:
> Can you add an example showing the recommended pattern of use?

One pattern is for example when reading from a stream and closing it is highly unlikely to generate an exception, then instead of the try block in
```
InputStream in = ...;
while(someCondition) {
    int b = in.read();
    // process b
}
try {
    in.close();
} catch (IOException ignored) {
}
```
one could simply write
```
in.closeUnchecked();
```
and avoid the boilerplate imposed by the checked exception. This sort of code can be found in many places in the tests in `<OpenJDK>/test/jdk`.

Reply:
OK, but note that using `closeUnchecked` in this way changes the behavior. Instead of ignoring an `IOException` it will get rethrown wrapped in an `UncheckedIOException`. This is probably reasonable for test code, where it's often acceptable to let exceptions propagate instead of handling them.

Are there other places in the JDK (in library code or in the tests) where `closeUnchecked` can and should be used? If so, maybe they should be updated as well.

Reply:
> This is probably reasonable for test code, where it's often acceptable to let exceptions propagate instead of handling them.

Understood.

> Are there other places in the JDK (in library code or in the tests) where `closeUnchecked` can and should be used? If so, maybe they should be updated as well.

Unless it's a matter of further background investigation, I think actually making any such changes would be better handled as a separate issue.

Reply:
> Unless it's a matter of further background investigation, I think actually making any such changes would be better handled as a separate issue.

OK. It doesn't have to be part of this change, but it would good if some example use cases could be identified. All too often I come up with great ideas for APIs, and then when I look around for example use sites, I can't find any ... and maybe this is telling me the API idea isn't as great as I thought it was.

Reply:
> and then when I look around for example use sites, I can't find any ... and maybe this is telling me the API idea isn't as great as I thought it was.

I hear you.